src/useQuery.ts

```diff
@@ -197,16 +199,45 @@ export function useQuery<
           setResponseId(x => x + 1);
         });
       };
-      const subscription = observableQuery.subscribe(
-        invalidateCurrentResult,
-        invalidateCurrentResult
-      );
+
+      // from: https://github.com/apollographql/react-apollo/blob/master/src/Query.tsx#L363
+      // after a error on refetch, without this fix, refetch never works again
+      function invalidateErrorResult() {
+        unsubscribe();
+
+        const lastError = observableQuery.getLastError();
+        const lastResult = observableQuery.getLastResult();
+
+        if (!suspend) {
```

Comment:
without this suspend test would fails, I'm not sure if it will breaks something.